.github/workflows/ci.yml

```diff
@@ -974,15 +974,30 @@ jobs:
       - boxes-test
       # - protocol-circuits-gates-report # non-blocking
     if: always()
+    outputs:
+      failure: ${{ steps.set_failed_jobs.outputs.failure }}
+      failed_jobs: ${{ steps.set_failed_jobs.outputs.failed_jobs }}
     steps:
+      - name: Check for Failures and Set Output
+        id: set_failed_jobs
+        env:
+          # Collect failed jobs from needs
+          FAILED_JOBS_JSON: ${{ toJson(needs) }}
```

Comment:
Are these actually the failed jobs or just all jobs that this one "needs"? If the latter then the comment and variable name are misleading.

Based on the run rule below, it looks like it's the latter, and the run rule extracts the ones that failed. But also I don't really know what I'm looking at!

Reply:
Good call, this is poorly named